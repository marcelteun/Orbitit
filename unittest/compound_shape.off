OFF
#
# SimpleShape
#
# file generated with python script by Marcel Tunnissen
# Vertices Faces Edges
16 14 0
# Vertices
1 1 1
-1 1 1
-1 -1 1
1 -1 1
1 1 -1
-1 1 -1
-1 -1 -1
1 -1 -1
1 1 -2
0.5 1 1
0 0 2
2 0 0
0 2 0
-2 0 0
0 -2 0
0 0 -2
# Sides and colours
4  0 1 2 3  204 25 25
4  0 3 7 4  204 25 25
4  1 0 4 5  204 25 25
4  2 1 5 6  204 25 25
4  3 2 6 7  204 25 25
4  7 6 5 4  204 25 25
3  10 11 12  204 204 76
3  10 12 13  204 204 76
3  10 13 14  204 204 76
3  10 14 11  204 204 76
3  15 12 11  204 204 76
3  15 13 12  204 204 76
3  15 14 13  204 204 76
3  15 11 14  204 204 76
# END
