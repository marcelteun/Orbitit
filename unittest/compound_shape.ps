%!PS-Adobe-2.0
%%Title: SimpleShape
%%Creator: PS.py written by marcelteun
%%Pages: 1 1
%%BoundingBox: 0 0 595 842
%%EndComments
%%BeginSetup
%%EndSetup
%%Page: 1 1
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [1 1] % 0
  [-1 1] % 1
  [-1 -1] % 2
  [1 -1] % 3
  [1 0] % 4
  [0 1] % 5
  [0 1] % 6
  [-1 0] % 7
  [-1 0] % 8
  [0 -1] % 9
  [0 -1] % 10
  [1 0] % 11
] def
/faces [
  [0 1 2 3][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [-1 1] % 0
  [-1 -1] % 1
  [1 1] % 2
  [1 -1] % 3
  [0 1] % 4
  [-1 0] % 5
  [-1 0] % 6
  [0 -1] % 7
  [1 0] % 8
  [0 1] % 9
  [0 -1] % 10
  [1 0] % 11
] def
/faces [
  [0 1 3 2][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [1 -1] % 0
  [-1 -1] % 1
  [1 1] % 2
  [-1 1] % 3
  [0 -1] % 4
  [1 0] % 5
  [-1 0] % 6
  [0 -1] % 7
  [1 0] % 8
  [0 1] % 9
  [0 1] % 10
  [-1 0] % 11
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [1 1] % 0
  [1 -1] % 1
  [-1 1] % 2
  [-1 -1] % 3
  [1 0] % 4
  [0 1] % 5
  [0 -1] % 6
  [1 0] % 7
  [0 1] % 8
  [-1 0] % 9
  [-1 0] % 10
  [0 -1] % 11
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [-1 1] % 0
  [1 1] % 1
  [-1 -1] % 2
  [1 -1] % 3
  [0 1] % 4
  [-1 0] % 5
  [1 0] % 6
  [0 1] % 7
  [-1 0] % 8
  [0 -1] % 9
  [0 -1] % 10
  [1 0] % 11
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [1 1] % 0
  [-1 1] % 1
  [-1 -1] % 2
  [1 -1] % 3
  [0 1] % 4
  [1 0] % 5
  [-1 0] % 6
  [0 1] % 7
  [0 -1] % 8
  [-1 0] % 9
  [1 0] % 10
  [0 -1] % 11
] def
/faces [
  [3 2 1 0][4 5][6 7][8 9][10 11]
] def
/bbox [-1 -1 1 1] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.000000 0 translate
/vertices [
  [-1.1547005 -1.1547005] % 0
  [1.5773503 -0.4226497] % 1
  [-0.4226497 1.5773503] % 2
  [-0.7886751 0.2113249] % 3
  [0.2113249 -0.7886751] % 4
  [0.2113249 -0.7886751] % 5
  [0.5773503 0.5773503] % 6
  [0.5773503 0.5773503] % 7
  [-0.7886751 0.2113249] % 8
] def
/faces [
  [0 1 2][3 4][5 6][7 8]
] def
/bbox [-1.1547005 -1.1547005 1.5773503 1.5773503] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [1.1547005 -1.1547005] % 0
  [0.4226497 1.5773503] % 1
  [-1.5773503 -0.4226497] % 2
  [-0.2113249 -0.7886751] % 3
  [0.7886751 0.2113249] % 4
  [0.7886751 0.2113249] % 5
  [-0.5773503 0.5773503] % 6
  [-0.5773503 0.5773503] % 7
  [-0.2113249 -0.7886751] % 8
] def
/faces [
  [0 1 2][3 4][5 6][7 8]
] def
/bbox [-1.5773503 -1.1547005 1.1547005 1.5773503] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [1.1547005 1.1547005] % 0
  [-1.5773503 0.4226497] % 1
  [0.4226497 -1.5773503] % 2
  [0.7886751 -0.2113249] % 3
  [-0.2113249 0.7886751] % 4
  [-0.2113249 0.7886751] % 5
  [-0.5773503 -0.5773503] % 6
  [-0.5773503 -0.5773503] % 7
  [0.7886751 -0.2113249] % 8
] def
/faces [
  [0 1 2][3 4][5 6][7 8]
] def
/bbox [-1.5773503 -1.5773503 1.1547005 1.1547005] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [-1.1547005 1.1547005] % 0
  [1.5773503 0.4226497] % 1
  [-0.4226497 -1.5773503] % 2
  [0.2113249 0.7886751] % 3
  [-0.7886751 -0.2113249] % 4
  [0.5773503 -0.5773503] % 5
  [0.2113249 0.7886751] % 6
  [-0.7886751 -0.2113249] % 7
  [0.5773503 -0.5773503] % 8
] def
/faces [
  [0 2 1][3 4][5 6][7 8]
] def
/bbox [-1.1547005 -1.5773503 1.5773503 1.1547005] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [0.4226497 -1.5773503] % 0
  [-1.5773503 0.4226497] % 1
  [1.1547005 1.1547005] % 2
  [-0.5773503 -0.5773503] % 3
  [0.7886751 -0.2113249] % 4
  [-0.2113249 0.7886751] % 5
  [-0.5773503 -0.5773503] % 6
  [0.7886751 -0.2113249] % 7
  [-0.2113249 0.7886751] % 8
] def
/faces [
  [2 1 0][3 4][5 6][7 8]
] def
/bbox [-1.5773503 -1.5773503 1.1547005 1.1547005] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [1.5773503 0.4226497] % 0
  [-0.4226497 -1.5773503] % 1
  [-1.1547005 1.1547005] % 2
  [0.5773503 -0.5773503] % 3
  [0.2113249 0.7886751] % 4
  [-0.7886751 -0.2113249] % 5
  [0.5773503 -0.5773503] % 6
  [0.2113249 0.7886751] % 7
  [-0.7886751 -0.2113249] % 8
] def
/faces [
  [2 1 0][3 4][5 6][7 8]
] def
/bbox [-1.1547005 -1.5773503 1.5773503 1.1547005] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [-0.4226497 1.5773503] % 0
  [1.5773503 -0.4226497] % 1
  [-1.1547005 -1.1547005] % 2
  [0.5773503 0.5773503] % 3
  [-0.7886751 0.2113249] % 4
  [0.2113249 -0.7886751] % 5
  [0.5773503 0.5773503] % 6
  [-0.7886751 0.2113249] % 7
  [0.2113249 -0.7886751] % 8
] def
/faces [
  [2 1 0][3 4][5 6][7 8]
] def
/bbox [-1.1547005 -1.1547005 1.5773503 1.5773503] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate
/vertices [
  [0.4226497 1.5773503] % 0
  [-1.5773503 -0.4226497] % 1
  [1.1547005 -1.1547005] % 2
  [0.7886751 0.2113249] % 3
  [-0.5773503 0.5773503] % 4
  [-0.5773503 0.5773503] % 5
  [-0.2113249 -0.7886751] % 6
  [-0.2113249 -0.7886751] % 7
  [0.7886751 0.2113249] % 8
] def
/faces [
  [2 0 1][3 4][5 6][7 8]
] def
/bbox [-1.5773503 -1.1547005 1.1547005 1.5773503] def
/scalingSize 1 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
7.732051 0 translate

showpage
%%EndPage

%%Tailer
%%EOF