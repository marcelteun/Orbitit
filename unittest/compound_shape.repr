Geom3D.CompoundShape(
    simpleShapes=[
        Geom3D.SimpleShape(
            Vs=[
                geomtypes.Vec3([1.0, 1.0, 1.0]),
                geomtypes.Vec3([-1.0, 1.0, 1.0]),
                geomtypes.Vec3([-1.0, -1.0, 1.0]),
                geomtypes.Vec3([1.0, -1.0, 1.0]),
                geomtypes.Vec3([1.0, 1.0, -1.0]),
                geomtypes.Vec3([-1.0, 1.0, -1.0]),
                geomtypes.Vec3([-1.0, -1.0, -1.0]),
                geomtypes.Vec3([1.0, -1.0, -1.0]),
                geomtypes.Vec3([1.0, 1.0, -2.0]),
                geomtypes.Vec3([0.5, 1.0, 1.0])
            ],
            Fs=[
                [0, 1, 2, 3],
                [0, 3, 7, 4],
                [1, 0, 4, 5],
                [2, 1, 5, 6],
                [3, 2, 6, 7],
                [7, 6, 5, 4]
            ],
            Es=[],
            colors=([(0.8, 0.1, 0.1)], [0, 0, 0, 0, 0, 0]),
            name="SimpleShape"
        ),
        Geom3D.SimpleShape(
            Vs=[
                geomtypes.Vec3([0.0, 0.0, 2.0]),
                geomtypes.Vec3([2.0, 0.0, 0.0]),
                geomtypes.Vec3([0.0, 2.0, 0.0]),
                geomtypes.Vec3([-2.0, 0.0, 0.0]),
                geomtypes.Vec3([0.0, -2.0, 0.0]),
                geomtypes.Vec3([0.0, 0.0, -2.0])
            ],
            Fs=[
                [0, 1, 2],
                [0, 2, 3],
                [0, 3, 4],
                [0, 4, 1],
                [5, 2, 1],
                [5, 3, 2],
                [5, 4, 3],
                [5, 1, 4]
            ],
            Es=[],
            colors=([(0.8, 0.8, 0.3)], [0, 0, 0, 0, 0, 0, 0, 0]),
            name="SimpleShape"
        )
    ],
    name="CompoundShape"
)
