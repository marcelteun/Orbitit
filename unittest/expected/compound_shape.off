OFF
#
# CompoundShape
#
# file generated with python script by Marcel Tunnissen
# Vertices Faces Edges
16 14 24
# Vertices
1 1 1
-1 1 1
-1 -1 1
1 -1 1
1 1 -1
-1 1 -1
-1 -1 -1
1 -1 -1
1 1 -2
0.5 1 1
0 0 2
2 0 0
0 2 0
-2 0 0
0 -2 0
0 0 -2
# Sides and colours
4  0 1 2 3  0.8 0.1 0.1
4  0 3 7 4  0.8 0.8 0.3
4  1 0 4 5  0.3 0.3 0.8
4  2 1 5 6  0.8 0.8 0.3
4  3 2 6 7  0.3 0.3 0.8
4  7 6 5 4  0.8 0.1 0.1
3  10 11 12  0.8 0.8 0.3
3  10 12 13  0.8 0.8 0.3
3  10 13 14  0.8 0.8 0.3
3  10 14 11  0.8 0.8 0.3
3  15 12 11  0.8 0.8 0.3
3  15 13 12  0.8 0.8 0.3
3  15 14 13  0.8 0.8 0.3
3  15 11 14  0.8 0.8 0.3
# END
