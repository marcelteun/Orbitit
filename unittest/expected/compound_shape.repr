orbitit.geom_3d.CompoundShape(
    shapes=[
        orbitit.geom_3d.SimpleShape(
            vs=[
                orbitit.geomtypes.Vec3([1, 1, 1]),
                orbitit.geomtypes.Vec3([-1, 1, 1]),
                orbitit.geomtypes.Vec3([-1, -1, 1]),
                orbitit.geomtypes.Vec3([1, -1, 1]),
                orbitit.geomtypes.Vec3([1, 1, -1]),
                orbitit.geomtypes.Vec3([-1, 1, -1]),
                orbitit.geomtypes.Vec3([-1, -1, -1]),
                orbitit.geomtypes.Vec3([1, -1, -1]),
                orbitit.geomtypes.Vec3([1, 1, -2]),
                orbitit.geomtypes.Vec3([0.5, 1, 1])
            ],
            Fs=[
                [0, 1, 2, 3],
                [0, 3, 7, 4],
                [1, 0, 4, 5],
                [2, 1, 5, 6],
                [3, 2, 6, 7],
                [7, 6, 5, 4]
            ],
            Es=[0, 1, 1, 2, 2, 3, 0, 3, 3, 7, 4, 7, 0, 4, 4, 5, 1, 5, 5, 6, 2, 6, 6, 7],
            colors=([(0.8, 0.1, 0.1), (0.8, 0.8, 0.3), (0.3, 0.3, 0.8)], [0, 1, 2, 1, 2, 0]),
            name="SimpleShape"
        ),
        orbitit.geom_3d.SimpleShape(
            vs=[
                orbitit.geomtypes.Vec3([0, 0, 2]),
                orbitit.geomtypes.Vec3([2, 0, 0]),
                orbitit.geomtypes.Vec3([0, 2, 0]),
                orbitit.geomtypes.Vec3([-2, 0, 0]),
                orbitit.geomtypes.Vec3([0, -2, 0]),
                orbitit.geomtypes.Vec3([0, 0, -2])
            ],
            Fs=[
                [0, 1, 2],
                [0, 2, 3],
                [0, 3, 4],
                [0, 4, 1],
                [5, 2, 1],
                [5, 3, 2],
                [5, 4, 3],
                [5, 1, 4]
            ],
            Es=[0, 1, 1, 2, 0, 2, 2, 3, 0, 3, 3, 4, 0, 4, 1, 4, 2, 5, 1, 5, 3, 5, 4, 5],
            colors=([(0.8, 0.8, 0.3)], [0, 0, 0, 0, 0, 0, 0, 0]),
            name="SimpleShape"
        )
    ],
    name="CompoundShape"
)
