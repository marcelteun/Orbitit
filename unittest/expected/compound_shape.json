{"class": "compound_shape", "data": {"name": "abc", "shapes": [{"class": "shape", "data": {"cols": [[0.8, 0.1, 0.1], [0.8, 0.8, 0.3], [0.3, 0.3, 0.8]], "face_cols": [0, 1, 2, 1, 2, 0], "fs": [[0, 1, 2, 3], [0, 3, 7, 4], [1, 0, 4, 5], [2, 1, 5, 6], [3, 2, 6, 7], [7, 6, 5, 4]], "name": "SimpleShape", "vs": [[1, 1, 1], [-1, 1, 1], [-1, -1, 1], [1, -1, 1], [1, 1, -1], [-1, 1, -1], [-1, -1, -1], [1, -1, -1], [1, 1, -2], [0.5, 1, 1]]}}, {"class": "shape", "data": {"cols": [[0.8, 0.8, 0.3]], "face_cols": [0, 0, 0, 0, 0, 0, 0, 0], "fs": [[0, 1, 2], [0, 2, 3], [0, 3, 4], [0, 4, 1], [5, 2, 1], [5, 3, 2], [5, 4, 3], [5, 1, 4]], "name": "SimpleShape", "vs": [[0, 0, 2], [2, 0, 0], [0, 2, 0], [-2, 0, 0], [0, -2, 0], [0, 0, -2]]}}]}}